govc/vm/vapp/configure.go

```diff
@@ -0,0 +1,160 @@
+package vapp
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+var allowedIPProtocols = []string{
+	string(types.VAppIPAssignmentInfoProtocolsIPv4),
+	string(types.VAppIPAssignmentInfoProtocolsIPv6),
+}
+
+var allowedAllocationPolicies = []string{
+	string(types.VAppIPAssignmentInfoIpAllocationPolicyDhcpPolicy),
+	string(types.VAppIPAssignmentInfoIpAllocationPolicyTransientPolicy),
+	string(types.VAppIPAssignmentInfoIpAllocationPolicyFixedPolicy),
+	string(types.VAppIPAssignmentInfoIpAllocationPolicyFixedAllocatedPolicy),
+}
+
+var allowedAllocationSchemes = []string{
+	string(types.VAppIPAssignmentInfoAllocationSchemesDhcp),
+	string(types.VAppIPAssignmentInfoAllocationSchemesOvfenv),
+}
+
+var allowedOVFTransports = []string{
+	"iso",
+	"com.vmware.guestInfo",
+}
+
+const configureCmdDescription = `Configures general vApp options on a virtual machine.
+
+List values are separated by commas.
+
+To remove vApp product properties, specify -product.remove. All relevant
+product properties need to be re-added after this operation - any product
+properties added at the same time are ignored.
+
+Examples:
+
+govc vm.vapp.configure -vm=foobar -ip.protocols.supported="IPv4,IPv6"
+govc vm.vapp.configure -vm=foobar -product.remove=true
+govc vm.vapp.configure -vm=foobar -product.name="Foo Bar"
+govc vm.vapp.configure -vm=foobar -product.name="Foo Bar" -product.version="1.0.0"
+
+`
+
+type configure struct {
+	*flags.VirtualMachineFlag
+
+	types.VmConfigSpec
+
+	removeProduct bool
+}
+
+func init() {
+	cli.Register("vm.vapp.configure", &configure{})
```

Comment:
Other Reconfigure_Task based commands are named "change", `vm.change`, `cluster.change`, etc.  I wouldn't object to changing that name (no pun intended) in the future, but would prefer to keep it consistent for now.